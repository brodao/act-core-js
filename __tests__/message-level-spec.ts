import log from '../src/logger';

test('Informação simples', () => {
	expect(log.gray('Informação simples')).toMatchSnapshot();
});

test('Informação com vários agumentos', () => {
	expect(
		log.gray('Informação com argumentos', 'arg 1', 'arg 2', 'arg 3')
	).toMatchSnapshot();
});

test('Aviso simples', () => {
	expect(log.warn('Aviso simples')).toMatchSnapshot();
});

test('Aviso com vários agumentos', () => {
	expect(
		log.warn('Aviso com argumentos', 'arg 1', 'arg 2', 'arg 3')
	).toMatchSnapshot();
});

test('Erro simples', () => {
	expect(log.error('Erro simples')).toMatchSnapshot();
});

test('Erro com vários agumentos', () => {
	expect(
		log.error('Erro com argumentos', 'arg 1', 'arg 2', 'arg 3')
	).toMatchSnapshot();
});

test('Verbose simples', () => {
	log.verboseOn();

	expect(log.verbose('Verbose simples')).toMatchSnapshot();
});

test('Verbose com vários agumentos', () => {
	log.verboseOff();

	expect(
		log.verbose('Informação com argumentos', {
			'key 1': 'arg 1',
			'key 2': 'arg 2',
			'key 3': 3,
		})
	).toMatchSnapshot();
});

test('Verbose desligado', () => {
	log.verboseOff();
	expect(log.verbose('Verbose desligado')).toMatchSnapshot();
});

test('Log com argumentos {}', () => {
	log.verboseOff();
	expect(
		log.gray('Log com argumentos {0}, {1}, {2}', 'arg 1', 'arg 2', 3)
	).toMatchSnapshot();
});
