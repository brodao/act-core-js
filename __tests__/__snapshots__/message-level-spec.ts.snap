// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aviso com vários agumentos 1`] = `undefined`;

exports[`Aviso simples 1`] = `undefined`;

exports[`Erro com vários agumentos 1`] = `undefined`;

exports[`Erro simples 1`] = `undefined`;

exports[`Informação com vários agumentos 1`] = `undefined`;

exports[`Informação simples 1`] = `undefined`;

exports[`Log com argumentos {} 1`] = `undefined`;

exports[`Verbose com vários agumentos 1`] = `undefined`;

exports[`Verbose desligado 1`] = `undefined`;

exports[`Verbose simples 1`] = `undefined`;
